### PR TITLE
UI specialist profile screen

### PR DESCRIPTION
**Pull request by romigben**
### Pull Request

---------------------------------------------

#### Comments

### CODE REVIEW

File: app/src/main/java/edu/uc/reedws/musiclink/ProfileActivity.kt [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.content.ContextCompat
+import com.google.android.material.floatingactionbutton.FloatingActionButton
+import kotlinx.android.synthetic.main.main_activity.bottomNav
+import kotlinx.android.synthetic.main.profile_view.*
+
+class ProfileActivity : AppCompatActivity() {
+
+    private val CAMERA_PERMISSION_REQUEST_CODE = 2000
```
Comment (reediculous456) [inline app/src/main/java/edu/uc/reedws/musiclink/ProfileActivity.kt:19]:
```suggestion
    private const val CAMERA_PERMISSION_REQUEST_CODE = 2000
```

File: app/src/main/java/edu/uc/reedws/musiclink/ProfileActivity.kt [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    private val CAMERA_REQUEST_CODE = 2001
```
Comment (reediculous456) [inline app/src/main/java/edu/uc/reedws/musiclink/ProfileActivity.kt:20]:
```suggestion
    private const val CAMERA_REQUEST_CODE = 2001
```

File: app/src/main/java/edu/uc/reedws/musiclink/ProfileActivity.kt [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+            takePhoto()
+        } else {
+            val permissionRequest = arrayOf(Manifest.permission.CAMERA)
+            requestPermissions(permissionRequest, CAMERA_PERMISSION_REQUEST_CODE)
+        }
+    }
+
+    override fun onRequestPermissionsResult(
+        requestCode: Int,
+        permissions: Array<out String>,
```
Comment (reediculous456) [inline app/src/main/java/edu/uc/reedws/musiclink/ProfileActivity.kt]: What does the `out` do here?

File: app/src/main/java/edu/uc/reedws/musiclink/ProfileActivity.kt [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+        changeNameDialogBuilder.setView(dialogLayout)
+        changeNameDialogBuilder.setPositiveButton("Done") { _, _ ->
+            txtPersonName.text = newName.text
+            val intent = Intent(this, ProfileActivity::class.java)
+            intent.putExtra("profileName", txtPersonName.text)
+        }
+        changeNameDialogBuilder.setNeutralButton("Cancel") { dialog, _ -> dialog.cancel() }
+        changeNameDialogBuilder.show()
+    }
```
Comment (reediculous456) [inline app/src/main/java/edu/uc/reedws/musiclink/ProfileActivity.kt:142]: Could we make these 3 dialogs 1 function that takes a couple parameters? feels like a lot of duplicated code

File: app/src/main/res/layout/profile_view.xml [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+        android:layout_width="wrap_content"
+        android:layout_height="75dp"
+        android:layout_marginTop="30dp"
+        android:orientation="horizontal">
+
+        <TextView
+            android:id="@+id/txtEmailAddress"
+            android:layout_width="304dp"
+            android:layout_height="wrap_content"
+            android:autofillHints="Enter your email address here..."
```
Comment (reediculous456) [inline app/src/main/res/layout/profile_view.xml:89]: This should be factored to `@strings`

File: app/src/main/res/layout/profile_view.xml [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+        android:layout_width="match_parent"
+        android:layout_height="75dp"
+        android:layout_marginTop="30dp"
+        android:orientation="horizontal">
+
+        <TextView
+            android:id="@+id/txtPersonName"
+            android:layout_width="304dp"
+            android:layout_height="63dp"
+            android:autofillHints="Enter your profile name here..."
```
Comment (reediculous456) [inline app/src/main/res/layout/profile_view.xml:60]: This should be factored to `@strings`

**Review by reediculous456 — CHANGES_REQUESTED** [review]

File: app/src/main/java/edu/uc/reedws/musiclink/ProfileActivity.kt (hunk 3 above)
Comment (romigben) [reply app/src/main/java/edu/uc/reedws/musiclink/ProfileActivity.kt]: Not sure, I was following the prof's tutorial. I removed it and it seems to work fine.

File: app/src/main/res/layout/profile_view.xml (hunk 5 above)
Comment (romigben) [reply app/src/main/res/layout/profile_view.xml:89]: This is factored in strings.xml. It shows up in gray text and when I click on it, it shows that it is already factored.

File: app/src/main/res/layout/profile_view.xml (hunk 6 above)
Comment (romigben) [reply app/src/main/res/layout/profile_view.xml:60]: This is factored in strings.xml. It shows up in gray text and when I click on it, it shows that it is already factored.

File: app/src/main/java/edu/uc/reedws/musiclink/ProfileActivity.kt (hunk 4 above)
Comment (romigben) [reply app/src/main/java/edu/uc/reedws/musiclink/ProfileActivity.kt:142]: For the sake of time, I can't get this to all be one method. There's a few things that have to be referenced and I'm not sure how to set them as parameters.

Comment (reediculous456) [reply app/src/main/java/edu/uc/reedws/musiclink/ProfileActivity.kt:142]: Ok sounds good

**Review by reediculous456 — DISMISSED** [review]

**Pull request merged**